pkg/management/postgres/instance_replica.go

```diff
@@ -34,30 +34,29 @@ func (instance *Instance) RefreshReplicaConfiguration(
 	cluster *apiv1.Cluster,
 	cli client.Client,
 ) (changed bool, err error) {
-	// The "archive_mode" setting was used to be overridden in the "postgresql.auto.conf"
-	// if the server was a designated primary. We need make sure to remove it
-	// and fall back on the value defined in "custom.conf".
-	// TODO: Removed this code together the RemoveArchiveModeFromPostgresAutoConf function
-	// TODO: when enough time passed since 1.12 release
-	changed, err = removeArchiveModeFromPostgresAutoConf(instance.PgData)
+	// TODO: Remove this code when enough time has passed since 1.21 release
+	//       This is due to the operator switching from postgresql.auto.conf
+	//       to override.conf for coordinating replication configuration
+	changed, err = migratePostgresAutoConfFile(ctx, instance, true)
```

Comment:
@litaocdl @YanniHu1996 
Can't we assume that if the new file exists, the migration is not needed? This would avoid running the code when not needed.

Reply:
Good suggestion! In the case of inplace upgrade, i think yes, we can leverage the existance of override.conf to skip the migration. for the. restore case, probably difficult as the file is created as empty in temp directory. but the two case can easy split, I will try.

Reply:
@armru I made a new commit which skip the migration by checking if the override.conf exist or not. this is for upgrade test case. for migration case, as override.conf is created in advance and only migrate one, I will let it force migrate, but migration will also be ignored if there is no content in postgresql.auto.conf (which means backup is already based on override.conf)